src/contexts/auth-context.js

```diff
@@ -0,0 +1,38 @@
+import { createContext, useContext, useState, useEffect } from 'react';
+
+const AuthContext = createContext();
+
+const authInitailState = { isAuthenticated: false, user: undefined };
+
+const AuthProvider = ({ children }) => {
+  const [authState, setAuth] = useState(authInitailState);
```

Comment:
Try to have same names for your state as well as setter function for easy identification. Like here it could have been:-
const [authState, setAuthState] = useState(authInitailState);